### PR TITLE
Adding basic support for margin endpoints

### PR DESCRIPTION
**Pull request by A-ndy-git**
Hello!

Due to the nature of Binance margin heavily overlapping with existing spot types; I have created this basic PR to verify if my approach is desired/correct.

Welcoming any comments/suggestions/changes.

Happy to add the rest of the margin types and functions once confirmed.

### CODE REVIEW

**Review by tiagosiebler — APPROVED** [review]
This looks good to me, just be careful please in case any `number` types should actually be `numberInString`. I'm not a fan of that mixed type but I'm not clear on a better way to represent a string containing a number (which can be converted to a number by the beautifier). Do you want me to merge this or do you wish to expand this first?

**Comment by A-ndy-git** [comment]
Thanks @tiagosiebler. I'll update those types to `numberInString`.

I have a strategy requiring this capability, so let me build in the rest of the endpoints and submit for review.

**Comment by A-ndy-git** [comment]
Hi @tiagosiebler. Most of the margin endpoints have been added; I have placed a TODO note in `Main-Client.ts` where these will be added at a later date. I should be able to pick this up later this week.

How does the proposed additions look? Happy to merge once you've signed off.

File: src/types/shared.ts
Comment (A-ndy-git) [reply src/types/shared.ts]: Makes sense @tiagosiebler. I've moved it across.

Anymore changes, or are you happy to merge?

File: src/types/shared.ts
Comment (tiagosiebler) [inline src/types/shared.ts:142]: StringBoolean?

**Review by tiagosiebler — APPROVED** [review]
Looks good - will merge this and push new version. If the param in CancelOCOParams should be StringBoolean, it can be sorted in the next PR. Thanks for all the work!

**Pull request merged**
